docs/topics/object-declarations.md

```diff
@@ -294,65 +427,125 @@ class MyClass {
-object of the class (whether named or not):
+```kotlin
+class User(val name: String) {
+    companion object {
+        private val defaultGreeting = "Hello"
+    }
+
+    fun sayHi() {
+        println(defaultGreeting)
+    }
+}
+User("Nick").sayHi()
+// Hello
+```
+
+When a class name is used by itself, it acts as a reference to the companion object of the class,
+regardless of whether the companion object is named or not:
 
 ```kotlin
-class MyClass1 {
-    companion object Named { }
+//sampleStart
+class User1 {
+    // Defines a named companion object
+    companion object Named {
+        fun show(): String = "User1's Named Companion Object"
+    }
 }
 
-val x = MyClass1
+// References the companion object of User1 using the class name
+val reference1 = User1
 
-class MyClass2 {
-    companion object { }
+class User2 {
+    // Defines an unnamed companion object
+    companion object {
+        fun show(): String = "User2's Companion Object"
+    }
 }
 
-val y = MyClass2
+// References the companion object of User2 using the class name
+val reference2 = User2
+//sampleEnd
+
+fun main() {
+    // Calls the show method from the companion object of User1
+    println(reference1.show()) 
+    // User1's Named Companion Object
+
+    // Calls the show method from the companion object of User2
+    println(reference2.show()) 
+    // User2's Companion Object
+}
 ```
+{kotlin-runnable="true"}
 
-Note that even though the members of companion objects look like static members in other languages, at runtime those
-are still instance members of real objects, and can, for example, implement interfaces:
+Although members of companion objects in Kotlin look like static members from other languages,
+they are actually instance members of the companion object. This allows companion objects to implement interfaces:
 
 ```kotlin
 interface Factory<T> {
-    fun create(): T
+    fun create(name: String): T
 }
 
-class MyClass {
-    companion object : Factory<MyClass> {
-        override fun create(): MyClass = MyClass()
+class User(val name: String) {
+    // Defines a companion object that implements the Factory interface
+    companion object : Factory<User> {
+        override fun create(name: String): User = User(name)
     }
 }
 
-val f: Factory<MyClass> = MyClass
+fun main() {
+    // Uses the companion object as a Factory
+    val userFactory: Factory<User> = User
+    val newUser = userFactory.create("Example User")
+    println(newUser.name)
+    // Example User
+}
 ```
+{kotlin-runnable="true"}
 
-However, on the JVM you can have members of companion objects generated as real static methods and fields if you use
+However, on the JVM, you can have members of companion objects generated as real static methods and fields if you use
 the `@JvmStatic` annotation. See the [Java interoperability](java-to-kotlin-interop.md#static-fields) section
 for more detail.
 
 ### Semantic difference between object expressions and declarations
```

Comment:
I think this should be one level higher 🤔 I'm wondering if it would be better to break up this section and add it to the relevant sections before. You can use these bits of information to motivate why this makes expressions/declarations so useful for certain use cases. WDYT?

Reply:
I think it's a very good idea to include it one level higher (makes far more sense). 👍 

Hmm 🤔 I think it's warranted to highlight it in a separate section. It draws more attention that way (it might get lost if we include this info in already information-heavy sections.

Reply:
OK, let's see how it looks once you've made the other changes :)